### PR TITLE
Support PHP 7.4 and Magento 2.4

### PR DESCRIPTION
**Pull request by andreustimm**

### CODE REVIEW

**Comment by Diego-Sotelo** [comment]
Hi,
Is there any prediction to work on PHP 7.4 and magento 2.4? Can we help in any way?
Thanks.

File: view/frontend/web/js/view/payment/method-renderer/boleto-method.js [hunk 1]
```diff
@@ -101,17 +101,19 @@ define(
                         'boleto_hash': PagSeguroDirectPayment.getSenderHash()
                     }
                 })).done(function () {
-                        delete paymentData['title'];                        
-                        $.when(placeOrder(paymentData, messageContainer)).done(function () {
-                          $.mage.redirect(window.checkoutConfig.pagseguro_boleto);
-                        });
+                    delete paymentData['title'];
+                    delete paymentData['__disableTmpl'];
```
Comment (andreustimm) [inline view/frontend/web/js/view/payment/method-renderer/boleto-method.js:105]: This line resolve the support of Magento 2.4

File: view/frontend/web/js/view/payment/method-renderer/credit_card-method.js [hunk 2]
```diff
@@ -159,10 +159,11 @@ define(
                 }
               })).done(function () {
                   delete paymentData['title'];
+                  delete paymentData['__disableTmpl'];
```
Comment (andreustimm) [inline view/frontend/web/js/view/payment/method-renderer/credit_card-method.js:162]: This line resolve the support of Magento 2.4

File: view/frontend/web/js/view/payment/method-renderer/default_lightbox-method.js [hunk 3]
```diff
@@ -61,14 +60,16 @@ define(
...
-                        delete paymentData['title'];
-                        $.when(placeOrder(paymentData, messageContainer)).done(function () {
-                           if (window.checkoutConfig.payment.pagseguro.isLightbox){
-                                $.mage.redirect(window.checkoutConfig.payment.pagseguro.checkout.lightbox);
-                            } else {
-                                $.mage.redirect(window.checkoutConfig.payment.pagseguro.checkout.standard);
-                            }
-                        });
+                    delete paymentData['title'];
+                    delete paymentData['__disableTmpl'];
```
Comment (andreustimm) [inline view/frontend/web/js/view/payment/method-renderer/default_lightbox-method.js:64]: This line resolve the support of Magento 2.4

File: view/frontend/web/js/view/payment/method-renderer/online_debit-method.js [hunk 4]
```diff
@@ -87,10 +87,12 @@ define(
                         'online_debit_bank' : self.checkedBank()
                     }
                 })).done(function () {
-                        delete paymentData['title'];
-                        $.when(placeOrder(paymentData, messageContainer)).done(function () {
-                          $.mage.redirect(window.checkoutConfig.pagseguro_boleto);
-                        });
+                    delete paymentData['title'];
+                    delete paymentData['__disableTmpl'];
```
Comment (andreustimm) [inline view/frontend/web/js/view/payment/method-renderer/online_debit-method.js:91]: This line resolve the support of Magento 2.4

**Review by andreustimm — COMMENTED** [review]
There is still a problem with this module, we need to create a label for **Multiple Payment**, else broke the Order in Admin.

**Comment by raphaelrejes** [comment]
Done

**Pull request merged**